src/components/forms/NetworkDevicesForm/NetworkDeviceIPAddress.tsx

```diff
@@ -0,0 +1,75 @@
+import { type FC } from "react";
+import type { InstanceAndProfileFormikProps } from "../instanceAndProfileFormValues";
+import { Input } from "@canonical/react-components";
+import { type LxdNetwork } from "types/network";
+import { type LxdNicDevice } from "types/device";
+import { typesWithNicStaticIPSupport } from "util/networks";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  index: number;
+  network?: LxdNetwork;
+  device: LxdNicDevice;
+  readOnly: boolean;
+  family: "IPv4" | "IPv6";
+}
+
+export const NetworkDeviceIPAddress: FC<Props> = ({
+  formik,
+  index,
+  network,
+  device,
+  readOnly,
+  family,
+}) => {
+  if (!network || !network.config) {
+    return null;
+  }
+
+  const addressString = family === "IPv4" ? "ipv4.address" : "ipv6.address";
+  const networkIP = network.config[addressString];
+  const deviceIP = device[addressString];
+  const isStaticIPAllowed =
+    network.managed && typesWithNicStaticIPSupport.includes(network.type);
+  const canSelectIP = isStaticIPAllowed && networkIP && networkIP != "none";
+
+  if (readOnly) {
+    return (
+      <div className="ip-content">
+        <>
+          <div>{family}</div>
+          <p className="u-no-margin--top u-no-margin--bottom">
+            {deviceIP || "Dynamic"}
+          </p>
+        </>
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <Input
+        id={`${addressString}-${index}`}
+        name={`${addressString}-${index}`}
+        label={`${family} address reservation`}
+        type="text"
+        placeholder={`Enter ${family} address`}
```

Comment:
I would really prefer the placeholder to be an IP address.
```suggestion
        const placeholder = networkIP?.split("/")[0] || `Enter ${family} address`;
        placeholder={placeholder}
```

<img width="428" height="361" alt="image" src="https://github.com/user-attachments/assets/2bde89bc-b827-42c3-932d-af74e9aa0056" />